pkg/server/datastore/sqlstore/sqlstore_test.go

```diff
@@ -2145,21 +2159,24 @@ func (s *PluginSuite) testListRegistrationEntries(dataConsistency datastore.Data
 					expectEntriesOut = [][]*common.RegistrationEntry{tt.expectEntriesOut}
 				}
 
-				var expectIDsOut [][]string
 				for _, entrySet := range expectEntriesOut {
-					var idSet []string
 					for _, entry := range entrySet {
-						idSet = append(idSet, entry.EntryId)
+						expectedEntriesOut[entry.EntryId] = entry
 					}
-					expectIDsOut = append(expectIDsOut, idSet)
 				}
 
 				if withPagination {
 					assert.Equal(t, tt.expectPagedTokensIn, tokensIn, "unexpected request tokens")
 				} else {
 					assert.Empty(t, tokensIn, "unexpected request tokens")
 				}
-				assert.Equal(t, expectIDsOut, actualIDsOut, "unexpected response entries")
+
+				assert.Len(t, actualEntriesOut, len(expectedEntriesOut), "unexpected number of entries returned")
+				for id, expectedEntry := range expectedEntriesOut {
+					// Some databases are not returning federated IDs in the same order (e.g. mysql)
```

Comment:
I've identified this behavior after adding a full message assertion. Should we consider it a bug?

Reply:
Hmm, certainly consistency would be good here, though shouldn't matter for correctness. It might through off some home-grown "compare these two entries for quality" code. We can probably just file an issue.

Reply:
I can open an issue for this 😄